storage/interceptable_repository.go

```diff
@@ -50,6 +50,7 @@ func newInterceptableRepository(repository Repository,
 	}
 }
 
+// InterceptableTransactionalRepository wraps a TransactionalRepository and holds sets of ordered interceptor providers for each object type
```

Comment:
holds sets of ordered interceptors why ? if you add documentation make sure it brings value to whoever reads it

Reply:
In hindsight, I see that this doc doesn't seem to bring any additional value than what the code itself says.
Do you have any better ideas or should I remove the doc?